tests/test_apply_factors.py

```diff
@@ -273,6 +279,15 @@ def test_method(self):
         ):
             mock_event = {
                 "json_data": json.loads(methodinput.to_json(orient="records")),
+                # Note, sum columns here actually overwrites 608 total
+                # but this way it checks both paths in sum_data_columns
+                "sum_columns": [{"column_name": "Q608_total", "data": {
+                                    "Q603_concreting_sand": "+",
+                                    "Q602_building_soft_sand": "+"}},
+                                {"column_name": "Q608_total", "data": {
+                                    "Q603_concreting_sand": "+",
+                                    "Q602_building_soft_sand": "-"}}
+                                ],
```

Comment:
If we use this to test both paths we should do different columns (Or can specify a new column I believe). Then add a second assert to test the output.

Reply:
This tests the paths. i added an extra set of variables to your test